core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -541,9 +541,12 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
 
     gcsDAO.listBillingAccounts(userInfo) flatMap { billingAccountNames =>
       billingAccountNames.find(_.accountName == billingAccountName) match {
-        case Some(billingAccount) if billingAccount.firecloudHasAccess => Future.successful(billingAccount)
         case None => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, createForbiddenErrorMessage("You", billingAccountName))))
-        case Some(billingAccount) if !billingAccount.firecloudHasAccess => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, createForbiddenErrorMessage(gcsDAO.billingEmail, billingAccountName))))
+        case Some(billingAccount) =>
+          if (billingAccount.firecloudHasAccess)
+            Future.successful(billingAccount)
+          else
+            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, createForbiddenErrorMessage(gcsDAO.billingEmail, billingAccountName))))
```

Comment:
this else clause is new - would like to hear from project/billing teams if this is the right error message for this case

Reply:
this is what @ehigham suggested